src/hotspot/share/compiler/compilerDirectives.hpp

```diff
@@ -185,6 +185,37 @@ class ControlIntrinsicIter {
   ControlIntrinsicIter& operator++();
 };
 
+class ControlIntrinsicValidator {
+ private:
+  bool _valid;
+  char* _bad;
+
+ public:
+  ControlIntrinsicValidator(ccstrlist option, bool disabled_all) : _valid(true), _bad(NULL) {
+    for (ControlIntrinsicIter iter(option, disabled_all); *iter != NULL && _valid; ++iter) {
+      if (vmIntrinsics::_none == vmIntrinsics::find_id(*iter)) {
+        _bad = NEW_C_HEAP_ARRAY(char, strlen(*iter) + 1, mtCompiler);
+        strncpy(_bad, *iter, strlen(*iter) + 1);
```

Comment:
This doesn't compile. Using strlen as an argument to strncpy is disallowed.

> "warning: 'char* __builtin_strncpy(char*, const char*, long unsigned int)' specified bound depends on the length of the source argument [-Wstringop-overflow=]"

Do a min between strlen and the maximum allowed length.

Fix this for both uses of the string length (row 197 and 198).

Reply:
Out of curiosity,   what kind of gcc do you use?  I am using gcc/g++-8.4.0 and I do append `--with-extra-cxxflags='-Wstringop-overflow -Wstringop-truncation'`, why can't I trigger this warning? 

I read this. https://developers.redhat.com/blog/2018/05/24/detecting-string-truncation-with-gcc-8/
let me try to fix it by replacing strncpy with strcpy.

Reply:
hi, @neliasso, 

I update my toolchain from g++-8 to g++-10 and successfully reproduce this issue. 
I have updated my code to fix it. Actually, only one line is modified. 
```
@@ -195,7 +195,7 @@ class ControlIntrinsicValidator {
     for (ControlIntrinsicIter iter(option, disabled_all); *iter != NULL && _valid; ++iter) {
       if (vmIntrinsics::_none == vmIntrinsics::find_id(*iter)) {
         _bad = NEW_C_HEAP_ARRAY(char, strlen(*iter) + 1, mtCompiler);
-        strncpy(_bad, *iter, strlen(*iter) + 1);
+        strcpy(_bad, *iter);
         _valid = false;
       }
     }
```

The change has passed all regression test. I don't know why the table above doesn't update. 
https://github.com/navyxliu/jdk/actions/runs/375913238

Could you take a look my patch, again?